src/input.ml

```diff
@@ -14,9 +14,12 @@ module type Key = sig
   type t [@@deriving repr]
   (** The type for keys. *)
 
+  (* The behaviour of the [encode] and [decode] functions is also available through repr in a much
+     more efficent way (i.e. copyless). *)
   val encode : t -> string
   (** [encode] is an encoding function. The resultant encoded values must have size {!encoded_size}. *)
 
+  (* (Almost) same: https://mirage.github.io/repr/repr/Repr/Size/index.html *)
   val encoded_size : int
   (** [encoded_size] is the size of the result of {!encode}, expressed in number of bytes. *)
```

Comment:
Hmm you're probably right. I remember that I was copying [what index does](https://github.com/mirage/index/blob/master/src/data.ml#L7-L35) but I'm not sure I have a god reason to do so.